ambry-protocol/src/main/java/com.github.ambry.protocol/DeleteRequest.java

```diff
@@ -27,27 +28,45 @@
  */
 public class DeleteRequest extends RequestOrResponse {
   private BlobId blobId;
+  private short accountId;
+  private short containerId;
+  private int deletionTimeInSecs;
   private int sizeSent;
-  private static final short Delete_Request_Version_V1 = 1;
+  static final short Delete_Request_Version_V1 = 1;
+  static final short Delete_Request_Version_V2 = 2;
+  private static short currentVersion = Delete_Request_Version_V1;
 
+  private static final int AccountId_ContainerId_Field_Size_InBytes = 2;
+  private static final int DeletionTime_Field_Size_InBytes = 4;
+
+  // @TODO: remove this constructor once DeleteRequest V2 is enabled
   public DeleteRequest(int correlationId, String clientId, BlobId blobId) {
-    super(RequestOrResponseType.DeleteRequest, Delete_Request_Version_V1, correlationId, clientId);
+    super(RequestOrResponseType.DeleteRequest, currentVersion, correlationId, clientId);
     this.blobId = blobId;
     sizeSent = 0;
   }
 
-  public static DeleteRequest readFrom(DataInputStream stream, ClusterMap map) throws IOException {
-    RequestOrResponseType type = RequestOrResponseType.DeleteRequest;
-    Short versionId = stream.readShort();
-    int correlationId = stream.readInt();
-    String clientId = Utils.readIntString(stream);
-    BlobId blobId = new BlobId(stream, map);
-    // ignore version for now
-    return new DeleteRequest(correlationId, clientId, blobId);
+  public DeleteRequest(int correlationId, String clientId, BlobId blobId, short accountId, short containerId,
+      int deletionTimeInSecs) {
+    super(RequestOrResponseType.DeleteRequest, Delete_Request_Version_V2, correlationId, clientId);
+    this.currentVersion = Delete_Request_Version_V2;
```

Comment:
you are changing a `static` variable here i.e. changing the value for all instances of this class. If you want to maintain an internal version, please have a `private` class member.